src/components/calltree/CallTree.js

```diff
@@ -240,31 +241,35 @@ class CallTreeComponent extends PureComponent<Props> {
 }
 
 export default explicitConnect<{||}, StateProps, DispatchProps>({
-  mapStateToProps: (state: State) => ({
-    threadIndex: getSelectedThreadIndex(state),
-    scrollToSelectionGeneration: getScrollToSelectionGeneration(state),
-    focusCallTreeGeneration: getFocusCallTreeGeneration(state),
-    tree: selectedThreadSelectors.getCallTree(state),
-    callNodeInfo: selectedThreadSelectors.getCallNodeInfo(state),
-    selectedCallNodeIndex: selectedThreadSelectors.getSelectedCallNodeIndex(
-      state
-    ),
-    rightClickedCallNodeIndex: selectedThreadSelectors.getRightClickedCallNodeIndex(
-      state
-    ),
-    expandedCallNodeIndexes: selectedThreadSelectors.getExpandedCallNodeIndexes(
-      state
-    ),
-    searchStringsRegExp: getSearchStringsAsRegExp(state),
-    disableOverscan: getPreviewSelection(state).isModifying,
-    invertCallstack: getInvertCallstack(state),
-    implementationFilter: getImplementationFilter(state),
-    icons: getIconsWithClassNames(state),
-    callNodeMaxDepth: selectedThreadSelectors.getCallNodeMaxDepth(state),
-    callTreeSummaryStrategy: selectedThreadSelectors.getCallTreeSummaryStrategy(
-      state
-    ),
-  }),
+  mapStateToProps: (state: State) => {
+    const callNodeInfo = getRightClickedCallNodeInfo(state);
+
+    return {
+      threadIndex: getSelectedThreadIndex(state),
+      scrollToSelectionGeneration: getScrollToSelectionGeneration(state),
+      focusCallTreeGeneration: getFocusCallTreeGeneration(state),
+      tree: selectedThreadSelectors.getCallTree(state),
+      callNodeInfo: selectedThreadSelectors.getCallNodeInfo(state),
+      selectedCallNodeIndex: selectedThreadSelectors.getSelectedCallNodeIndex(
+        state
+      ),
+      rightClickedCallNodeIndex: callNodeInfo
+        ? callNodeInfo.callNodeIndex
+        : null,
```

Comment:
for more correctness, maybe we should pass `null` if the selected threadIndex is different than the threadIndex in the callNodeInfo. What do you think?

Of course this is the same in other similar locations.
Because the code will be the same, maybe we should extract a simple function:
```js
function getRightClickedCallNodeIndexForThread(state, threadIndex) {
   const callNodeInfo = getRightClickedCallNodeInfo(state);
  if (!callNodeInfo || callNodeInfo.threadIndex !== threadIndex) {
    return null;
  }
  return callNodeInfo.callNodeIndex;
}
```
I'm thinking that maybe this function could even be in `selectedThreadSelectors`. This way we could still have `selectedThreadSelectors.getRightClickedCallNodeIndex`. (it would be slightly different, that it wouldn't take threadIndex as parameter).

What do you think? Would that work?
This is all to make the code more readable in the end.

Reply:
Yeah, good catch, this makes a lot of sense! Having a helper seems to be pretty straigtforward, but moving it to `selectedThreadSelectors` will cause a circular dependency between per-thread selectors and right-clicked selectors in the current implementation. I'll look at this case more closely and see if this can be resolved somehow 🤔